richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -1,49 +1,135 @@
 package com.halilibo.richtext.ui.string
 
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.tween
 import androidx.compose.foundation.layout.BoxWithConstraints
+import androidx.compose.foundation.text.InlineTextContent
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.invalidateGroupsWithKey
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateMapOf
+import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
 import androidx.compose.ui.text.TextLayoutResult
 import androidx.compose.ui.text.style.TextOverflow
 import com.halilibo.richtext.ui.ClickableText
 import com.halilibo.richtext.ui.RichTextScope
 import com.halilibo.richtext.ui.currentContentColor
 import com.halilibo.richtext.ui.currentRichTextStyle
 import com.halilibo.richtext.ui.string.RichTextString.Format
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.sample
+import kotlinx.coroutines.launch
+import kotlin.time.Duration.Companion.milliseconds
 
 /**
  * Renders a [RichTextString] as created with [richTextString].
  *
  * @sample com.halilibo.richtext.ui.previews.TextPreview
  */
+@OptIn(FlowPreview::class)
 @Composable
 public fun RichTextScope.Text(
   text: RichTextString,
   modifier: Modifier = Modifier,
   onTextLayout: (TextLayoutResult) -> Unit = {},
   softWrap: Boolean = true,
+  isLeafText: Boolean = true,
+  animate: Boolean = false,
+  textFadeInMs: Int = 600,
+  debounceMs: Int = 150,
   overflow: TextOverflow = TextOverflow.Clip,
   maxLines: Int = Int.MAX_VALUE
 ) {
   val style = currentRichTextStyle.stringStyle
   val contentColor = currentContentColor
+  val coroutineScope = rememberCoroutineScope()
   val annotated = remember(text, style, contentColor) {
     val resolvedStyle = (style ?: RichTextStringStyle.Default).resolveDefaults()
     text.toAnnotatedString(resolvedStyle, contentColor)
   }
+  val animations = remember { mutableStateMapOf<Int, TextAnimation>() }
+  val textToRender = remember { mutableStateOf(AnnotatedString("")) }
+
+  if (animate) {
```

Comment:
Maybe you could wrap up this block with a `rememberAnimatedText(…)` or something to keep it isolated?

Reply:
oh nice, agreed